core/shared/src/test/scala/com/monovore/decline/ArgumentSpec.scala

```diff
@@ -60,6 +63,53 @@ class ArgumentSpec extends ArgumentSuite {
   checkArgument[BigInt]("BigInt")
   checkArgument[UUID]("UUID")
 
+  val onlyValidArbitraryDuration = Arbitrary(
+    Gen.frequency(
+      1 -> Gen.const(Duration.Inf),
+      1 -> Gen.const(Duration.MinusInf),
+      1 -> Gen.const(Duration.Zero),
+      6 -> Gen.finiteDuration
+    )
+  )
+
+  val infinity1Show = Show.show[Duration] {
```

Comment:
If you check https://github.com/scala/scala/blob/8a2cf63ee5bad8c8c054f76464de0e10226516a0/src/library/scala/concurrent/duration/Duration.scala#L61-L63 you will see that there are multiple ways that the default `Duration.apply(s: String)` can parse negative/positive Infinity. To deal with these I have just created different variations of `cats.show.Show`  corresponding to the infinities.

Ideally I would have liked to have a single `cats.show.Show` that just uses the seed from Scalacheck and depending on that seed generates one of the various serializations of positive/negative Infinity however afaik its not possible to access the seed at this point (maybe with some refactoring of tests this is possible but if so that would be for another PR).

Reply:
Personally I'd also be comfortable with using the default `Show`/`toString` for all durations, and then adding a couple extra explicit test cases to cover the missing parts of the space. (As you can imagine this same issue crops up for floats as well... lots of ways to write down the same floating point number!)

Happy to keep this implementation too... but in that case, mind adding the context you shared here as a comment on the code? Might not be obvious to future readers why this is necessary.

Reply:
So the issue is that the default cats `Show` for `Duration` does not work because the `String` values it outputs for the various infinite representations are different from what `Duration.apply(s: String)` accepts.

Are you fine with me going with the option of commenting code to provide context instead?

Reply:
> So the issue is that the default cats Show for Duration does not work because the String values it outputs for the various infinite representations are different from what Duration.apply(s: String) accepts.

Right. In that case a `Show` that generates _any_ of the valid inputs for infinity would work here. (Note that we're not generating all of the possible valid inputs for finite durations either... `ms` vs. milliseconds etc.)

> Are you fine with me going with the option of commenting code to provide context instead?

Yep, also totally fine!

Reply:
Okay, I will create a single generic `Show` for `Duration` and then test the other infinite string representations explicitly, that should make it more clear

Reply:
Okay done, let me know if you are happy with it